src/main/java/com/google/devtools/build/lib/rules/repository/RepoRecordedInput.java

```diff
@@ -76,22 +78,51 @@ public abstract static class Parser {
    * Parses a recorded input from its string representation.
    *
    * @param s the string representation
-   * @param baseDirectory the path to a base directory that any filesystem paths should be resolved
-   *     relative to
    * @return The parsed recorded input object, or {@code null} if the string representation is
    *     invalid
    */
   @Nullable
-  public static RepoRecordedInput parse(String s, Path baseDirectory) {
+  public static RepoRecordedInput parse(String s) {
     List<String> parts = Splitter.on(':').limit(2).splitToList(s);
     for (Parser parser : new Parser[] {File.PARSER, EnvVar.PARSER, RecordedRepoMapping.PARSER}) {
       if (parts.get(0).equals(parser.getPrefix())) {
-        return parser.parse(parts.get(1), baseDirectory);
+        return parser.parse(parts.get(1));
       }
     }
     return null;
   }
 
+  /**
+   * Returns whether all values are still up-to-date for each recorded input. If Skyframe values are
+   * missing, the return value should be ignored; callers are responsible for checking {@code
+   * env.valuesMissing()} and triggering a Skyframe restart if needed.
+   */
+  public static boolean areAllValuesUpToDate(
+      Environment env, Map<? extends RepoRecordedInput, String> recordedInputValues)
+      throws InterruptedException {
+    var skyKeys = ImmutableSet.<SkyKey>builder();
+    for (RepoRecordedInput rri : recordedInputValues.keySet()) {
+      skyKeys.add(rri.getSkyKey(env));
+    }
+    env.getValuesAndExceptions(skyKeys.build());
+    if (env.valuesMissing()) {
+      return false;
```

Comment:
nit: it's kinda confusing that you signal "deps not available yet" and "not up to date" in the same way. I undestand that the call site calls `env.valuesMissing()`, but a comment would still be useful or even a ternary `Boolean` / `Optional<Boolean>` (instead of `boolean`) return value.

Reply:
the method javadoc says that callers are responsible for checking env.valuesMissing(). do you think that's enough?

Reply:
Works for me. I admit, though, that I didn't read the Javadoc which says either that I'm sloppy or that it's not as frequently read as you might think.